cmd/kubeadm/app/cmd/init.go

```diff
@@ -457,18 +459,105 @@ func (i *Init) Run(out io.Writer) error {
 	}
 
 	ctx := map[string]string{
-		"KubeConfigPath": adminKubeConfigPath,
 		"Token":          i.cfg.Token,
 		"CAPubKeyPin":    pubkeypin.Hash(caCert),
 		"MasterHostPort": masterHostPort,
 	}
+
+	kubeConfigInfo := fmt.Sprintf(`
+To start using your cluster, you need to run the following as a regular user:
+
+	  mkdir -p $HOME/.kube
+	  sudo cp -i %s $HOME/.kube/config
+	  sudo chown $(id -u):$(id -g) $HOME/.kube/config
+
+`, adminKubeConfigPath)
+
 	if i.skipTokenPrint {
 		ctx["Token"] = "<value withheld>"
 	}
 
+	ctx["KubeConfigInfo"] = kubeConfigInfo
+	if i.defaultUser != "" {
+		if err := copyConfigForDefaultUser(i.defaultUser, adminKubeConfigPath); err == nil {
+			ctx["KubeConfigInfo"] = ""
+		} else {
+			fmt.Printf("[init] WARNING: Could not copy the configuration for user %q: %v.\n", i.defaultUser, err)
+		}
+	}
+
 	return initDoneTempl.Execute(out, ctx)
 }
 
+func copyConfigForDefaultUser(defaultUser string, adminKubeConfigPath string) error {
+
+	var err error
+	var uid, gid int
+	var usr *user.User
+	var src, dest *os.File
+	var kubeDir, configFilePath string
+
+	fmt.Printf("[init] Copying configuration to %q from %q.\n", configFilePath, adminKubeConfigPath)
+
+	usr, err = user.Lookup(defaultUser)
+	if err != nil {
+		return fmt.Errorf("no such user")
+	}
+
+	if usr.HomeDir == "" {
+		return fmt.Errorf("cannot obtain the home directory for this user")
+	}
+	kubeDir = usr.HomeDir + "/.kube"
+
+	uid, err = strconv.Atoi(usr.Uid)
+	if err != nil {
+		return fmt.Errorf("cannot Atoi() UID string %q", usr.Uid)
+	}
+
+	gid, err = strconv.Atoi(usr.Gid)
+	if err != nil {
+		return fmt.Errorf("cannot Atoi() GID string %q", usr.Gid)
+	}
+
+	err = os.Mkdir(kubeDir, 0700)
+	// return if the error is not about the directory already existing
+	if err != nil {
+		if !os.IsExist(err) {
+			return fmt.Errorf("cannot create %q", kubeDir)
+		}
+	}
+
+	err = os.Chown(kubeDir, uid, gid)
```

Comment:
Is this chown recursive or not?

Reply:
TMK, no it is not. the problem here reading https://golang.org/pkg/os/#Chown seems to be that on Win32 it will always return an Error. in a way, the `chown` calls might have to be branched-out on Win32.